sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientBuilder.java

```diff
@@ -78,50 +85,55 @@ public NettyAsyncHttpClientBuilder(HttpClient nettyHttpClient) {
                     tcpConfig = tcpConfig.runOn(nioEventLoopGroup);
                 }
 
-                if (proxyOptions != null) {
-                    ProxyProvider.Proxy nettyProxy;
-                    switch (proxyOptions.getType()) {
-                        case HTTP:
-                            nettyProxy = ProxyProvider.Proxy.HTTP;
-                            break;
-                        case SOCKS4:
-                            nettyProxy = ProxyProvider.Proxy.SOCKS4;
-                            break;
-                        case SOCKS5:
-                            nettyProxy = ProxyProvider.Proxy.SOCKS5;
-                            break;
-                        default:
-                            throw logger.logExceptionAsError(new IllegalStateException(
-                                String.format("Unknown Proxy type '%s' in use. Not configuring Netty proxy.",
-                                    proxyOptions.getType())));
-                    }
-                    if (proxyOptions.getUsername() != null) {
-                        // Netty supports only Basic proxy authentication and we default to it.
-                        return tcpConfig.proxy(ts -> ts.type(nettyProxy)
-                                .address(proxyOptions.getAddress())
-                                .username(proxyOptions.getUsername())
-                                .password(userName -> proxyOptions.getPassword())
-                                .build());
-                    } else {
-                        return tcpConfig.proxy(ts -> ts.type(nettyProxy).address(proxyOptions.getAddress()));
-                    }
+                ProxyOptions buildProxyOptions = (proxyOptions == null)
+                    ? ProxyOptions.loadFromEnvironment(buildConfiguration)
```

Comment:
nit: `loadFromConfiguration` would be better a name as you are passing the configuration to this method (and configuration can load data from the environment or somewhere else)

Reply:
Renamed